enhancements/authentication/AuthConfig-missing-fields.md

```diff
@@ -0,0 +1,526 @@
+   - The updated configuration is applied to the cluster.
+   - OpenShift's `cluster-authentication-operator` processes the changes and propagates them to relevant components (e.g., `kube-apiserver`).
+
+3. **Validation and Enforcement**  
+   - The cluster authentication system ensures that the provided values are correctly interpreted and enforced.
+   - If `ClaimValidationRules` or `UserValidationRules` are configured, OpenShift evaluates authentication requests against them.
+
+4. **Authentication Flow Execution**  
+   - When a user authenticates, OpenShift uses the configured values to validate identity claims, apply rules, and determine user permissions.
+
+
+### Use Case Examples
+
+#### Precondition
+A customer has a subscription for an identity provider (IdP) that they would like their employees to use for authenticating with their OpenShift clusters. The IdP provides identity services but may have certain non-standard configurations that require customization within OpenShift.
+
+#### 1. Configuring a Non-Standard OIDC Discovery URL  
+**Scenario:**  
+A company is using an external identity provider to authenticate users across multiple services, including OpenShift. However, this IdP does not follow the standard OIDC discovery URL format. To integrate OpenShift with this provider, the cluster administrator must manually specify the custom discovery URL in OpenShift’s authentication configuration.  
+
+**Steps for the Cluster Administrator:**  
+1. Update the `Authentication` CRD to include the custom `discoveryURL`:  
+   ```yaml
+   apiVersion: config.openshift.io/v1
+   kind: Authentication
+   metadata:
+     name: cluster
+   spec:
+     oidcProviders:
+       - issuer:
+           discoveryURL: "https://custom-idp.example.com/.well-known/openid-configuration"
+    ```
+
+#### 2. Configuring `AudienceMatchPolicy` for Token Validation
+**Scenario:**  
+A cluster administrator needs to configure OpenShift to validate the audience (`aud`) claim in JWT tokens using a flexible matching policy.
+
+**Steps:**
+1. Update the authentication CRD with `AudienceMatchPolicy`:
+   ```yaml
+   apiVersion: config.openshift.io/v1
+   kind: Authentication
+   metadata:
+     name: cluster
+   spec:
+     audienceMatchPolicy: MatchAny
+     ```
+
+#### 3. Custom UID Mapping for Identity Providers
+**Scenario:**  
+A company needs to map the `uid` claim to a custom identifier in order to properly identify users across different OIDC providers.
+
+**Steps:**
+
+1. Update the authentication CRD to include the custom `uid` mapping:
+   ```yaml
+   apiVersion: config.openshift.io/v1
+   kind: Authentication
+   metadata:
+     name: cluster
+   spec:
+     tokenClaimMappings:
+       uid: claims.email  # Example of custom claim mapping
+  ```
+
+### 4. Adding Extra Claims for Role-Based Permissions
+**Scenario:**  
+A cluster administrator needs to map an extra claim (such as `role`) from the OIDC token for role-based access control (RBAC).
```

Comment:
I don't know that this is useful with the native Kubernetes RBAC authorization process. My understanding is that using the native RBAC resources there is no way to create a rule that would take into consideration the `UserInfo.Extras` field.

I would recommend adjusting this scenario a bit to include the use of a webhook authorizer that processes these custom extra fields.